src/consumer/component/consumer-component.js

```diff
@@ -481,38 +483,36 @@ export default class Component {
       };
 
       return installEnvironmentIfNeeded()
-      .then(() => {
-        if (!compiler) {
-          compiler = scope.loadEnvironment(this.compilerId);
-        }
-        // todo: what files should be built?
-        const buildFilesP = this.buildIfNeeded({
-          condition: !!this.compilerId,
-          compiler,
-          files: this.files,
-          consumer,
-          scope
-        });
-
-        return buildFilesP().then((buildedFiles) => {
-          buildedFiles.forEach((file) =>{
-            if (file && (!file.contents || !isString(file.contents.toString()))) {
-              throw new Error('builder interface has to return object with a code attribute that contains string');
-            }
+        .then(() => {
+          if (!compiler) {
+            compiler = scope.loadEnvironment(this.compilerId);
+          }
+          // todo: what files should be built?
+          const buildFilesP = this.buildIfNeeded({
+            condition: !!this.compilerId,
+            compiler,
+            files: this.files,
+            consumer,
+            scope
           });
 
-          this.dist = buildedImpl;
-          this.specDist = buildedSpec;
+          return Promise.resolve(buildFilesP).then((buildedFiles) => {
```

Comment:
The Promise.resolve is not needed here I think (make sure it still works after changing this).
It should be:
```js
buildFilesP().then((buildedFiles)
```